### PR TITLE
ci: fix expensive e2e tests

### PR DESCRIPTION
**Pull request by brojd**
wait until plugin backend is built before grafana starts via ops-devenv

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] Added the relevant release notes label (see labels prefixed w/ `release:`). These labels dictate how your PR will
    show up in the autogenerated release notes.

### CODE REVIEW

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -143,7 +143,7 @@ jobs:
         shell: bash
         run: |
           cd ..
-          git clone https://x-access-token:${{ steps.generate-token.outputs.token }}@github.com/grafana/ops-devenv.git
+          git clone https://x-access-token:${{ steps.generate-token.outputs.token }}@github.com/grafana/ops-devenv.git -b support-grafana-deps-append
```
Comment (brojd) [inline .github/workflows/e2e-tests.yml:146]: revert after merging https://github.com/grafana/ops-devenv/pull/21

File: .github/workflows/e2e-tests.yml [hunk 2]
```diff
@@ -168,7 +168,6 @@ jobs:
 
       - name: Tilt CI - standard E2E tests
         shell: bash
-        if: ${{ inputs.run-expensive-tests == false }}
```
Comment (joeyorlando) [inline .github/workflows/e2e-tests.yml:171]: ohhh so previously on the daily cron workflow we weren't running the "non-expensive" tests? 👀

File: .github/workflows/expensive-e2e-tests.yml [hunk 3]
```diff
@@ -14,21 +14,8 @@ jobs:
...
-          # 10.0.11 is the earliest version where things work
-          # - 9.2.20
-          # - 9.3.16
-          # - 9.4.13
-          # - 9.5.7
           - 10.1.7
           - 10.3.3
-          # TODO: fix issues with running e2e tests against Grafana v10.2.x and v10.3.x
-          # - 10.2.4
-          # - latest
```
[the remark this answers is not in the document]
Comment (brojd) [reply .github/workflows/expensive-e2e-tests.yml:31]: Totally, maybe let me create a dedicated issue for that in IRM project as we will be running e2e tests for all 3 plugins there?

File: .github/workflows/e2e-tests.yml (hunk 2 above)
Comment (brojd) [reply .github/workflows/e2e-tests.yml:171]: It was but it was hidden under npm command in package.json which I changed yesterday. Instead, now it's explicit in GH workflows - every time we run standard e2e tests and daily + main branch workflows will additionally run expensive ones

**Pull request merged**